schematics/utils/html.ts

```diff
@@ -0,0 +1,56 @@
+import {Tree, SchematicsException} from '@angular-devkit/schematics';
+import * as parse5 from 'parse5';
+import {getIndexPath} from './ast';
+import {InsertChange} from './devkit-utils/change';
+
+/**
+ * Parses the index.html file to get the HEAD tag position.
+ */
+export function getHeadTag(host: Tree, src: string) {
+  const document = parse5.parse(src,
+    { locationInfo: true }) as parse5.AST.Default.Document;
+
+  let head;
+  const visit = (nodes: parse5.AST.Default.Node[]) => {
+    nodes.forEach(node => {
+      const element = <parse5.AST.Default.Element>node;
+      if (element.tagName === 'head') {
+        head = element;
+      } else {
+        if (element.childNodes) {
+          visit(element.childNodes);
+        }
+      }
+    });
+  };
+
+  visit(document.childNodes);
+
+  if (!head) {
+    throw new SchematicsException('Head element not found!');
+  }
+
+  return {
+    position: head.__location.startTag.endOffset
```

Comment:
Why does this need to use a `__`-prefixed member?

Reply:
Unsure, this is just what the object looks like.

Reply:
This is typically meant to communicate a private API; there might be a alternate public way to access this

Reply:
Is there a reason you're using parse5 over domino? Domino might provide a more natural interface here since it more closely mirrors the DOM than parse5.

Reply:
That is a good point, especially since Angular Universal has moved to Domino. For now we can continue with this, though, since our use of it is pretty minor.